pkg/utils/mainhelper/options.go

```diff
@@ -54,6 +55,9 @@ func newInitArgs(component string, opts ...InitOption) initArgs {
 		// Set up signals so we handle the first shutdown signal gracefully.
 		args.ctx = signals.NewContext()
 	}
+	if args.metricNamespace == "" {
```

Comment:
Can you add a unit test for the new option? Otherwise lgtm

Reply:
Done